xarray/tests/test_accessor_dt.py

```diff
@@ -148,6 +148,118 @@ def test_rounders(self):
         assert_array_equal(dates.round("D").values, xdates.time.dt.round("D").values)
 
 
+class TestTimedeltaProperties:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        nt = 100
+        data = np.random.rand(10, 10, nt)
+        lons = np.linspace(0, 11, 10)
+        lats = np.linspace(0, 20, 10)
+        self.times = pd.date_range(start="2000/01/01", freq="H", periods=nt)
+        self.times = self.times - self.times[0]
+
+        self.data = xr.DataArray(
+            data,
+            coords=[lons, lats, self.times],
+            dims=["lon", "lat", "time"],
+            name="data",
+        )
+
+        self.times_arr = np.random.choice(self.times, size=(10, 10, nt))
+        self.times_data = xr.DataArray(
+            self.times_arr,
+            coords=[lons, lats, self.times],
+            dims=["lon", "lat", "time"],
+            name="data",
+        )
+
+    def test_field_access(self):
+        days = xr.DataArray(
+            self.times.days, name="days", coords=[self.times], dims=["time"]
+        )
+        seconds = xr.DataArray(
+            self.times.seconds, name="seconds", coords=[self.times], dims=["time"]
+        )
+        microseconds = xr.DataArray(
+            self.times.microseconds,
+            name="microseconds",
+            coords=[self.times],
+            dims=["time"],
+        )
+        nanoseconds = xr.DataArray(
+            self.times.nanoseconds,
+            name="nanoseconds",
+            coords=[self.times],
+            dims=["time"],
+        )
+
+        assert_equal(days, self.data.time.dt.days)
+        assert_equal(seconds, self.data.time.dt.seconds)
+        assert_equal(microseconds, self.data.time.dt.microseconds)
+        assert_equal(nanoseconds, self.data.time.dt.nanoseconds)
+
+    def test_not_datetime_type(self):
+        nontime_data = self.data.copy()
+        int_data = np.arange(len(self.data.time)).astype("int8")
+        nontime_data["time"].values = int_data
+        with raises_regex(TypeError, "dt"):
+            nontime_data.time.dt
+
+    @requires_dask
+    def test_dask_field_access(self):
+        import dask.array as da
+
+        days = self.times_data.dt.days
```

Comment:
could this be parameterized?

``` python
@pytest.mark.parametrize("field", ["seconds", ...])
...
...
expected = getattr(self.time_data.dt, field)
with raise_if_dask_computes():
    actual = getattr(dask_times_2d.dt, field)

assert isinstance(actual.data, dask.array.Array)
assert_chunks_equal(actual, dask_times_2d)
assert_equal(actual, expected)

```

Reply:
I was able to parametrize the properties. What about the methods, i.e (`floor()`, `ceil()`, `round()`) ?

Reply:
`getattr(Ri.time.dt, "ceil")("D")` works on master. Does it not work now?

Reply:
Thank you for the hint! I am now able to get it to work